pkg/cortex/modules.go

```diff
@@ -306,6 +311,7 @@ func (t *Cortex) stopQuerierChunkStore() error {
 
 func (t *Cortex) initIngester(cfg *Config) (err error) {
 	cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.Multi.ConfigProvider = multiClientRuntimeConfigChannel(t.runtimeConfig)
+	cfg.Ingester.LifecyclerConfig.RingConfig.KVStore.MemberlistKV = t.memberlistKVState.getMemberlistKV
```

Comment:
I think you will also have to inject this into the ruler and the distributor. The ruler runs a separate ring for rule evals and the distributor runs a separate ring for HA. Or are we going to assume people will set those rings up on separate ports/kv stores?